### PR TITLE
ARROW-334: [Python] Remove INSTALL_RPATH_USE_LINK_PATH

### PR DESCRIPTION
**Pull request by wesm**
Will try to verify whether this resolves the issue. See https://travis-ci.org/conda-forge/staged-recipes/builds/166897102

### CODE REVIEW

File: python/CMakeLists.txt [hunk 1]
```diff
@@ -417,8 +417,6 @@ if (UNIX)
   set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
 endif()
 
-SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
```
Comment (xhochy) [inline python/CMakeLists.txt]: Does this really have an effect. Reading the documentation for `CMAKE_BUILD_WITH_INSTALL_RPATH` above it seems like they affect each other a bit.

Comment (wesm) [inline python/CMakeLists.txt]: See the failing build -- the INSTALL_RPATH is being discarded

File: ci/travis_script_python.sh [hunk 2]
```diff
@@ -14,12 +14,16 @@
...
+
 PYTHON_DIR=$TRAVIS_BUILD_DIR/python
 
 # Re-use conda installation from C++
 export MINICONDA=$HOME/miniconda
 export PATH="$MINICONDA/bin:$PATH"
-export PARQUET_HOME=$MINICONDA
+
+export ARROW_HOME=$ARROW_CPP_INSTALL
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$ARROW_CPP_INSTALL
```
Comment (xhochy) [inline ci/travis_script_python.sh]: Not so happy that this is needed now but I can live with it as the `manylinux1`  `pyarrow` packages will not need it due to some separate post-processing.

Comment (wesm) [inline ci/travis_script_python.sh]: It's only for Travis. It was working before because of magic rpath CMake stuff

**Comment by wesm** [comment]
OK, now this reproduces the failure observed in https://github.com/conda-forge/staged-recipes/pull/1732, will try to debug locally

**Comment by wesm** [comment]
Alright, so it's clear, then, that XCode 6.1 is the problem. I also tested on XCode 8.0 and no problems with `@loader_path` there. I know way more than I ever wanted to about rpaths, but it's moderately useful, I suppose?

**Comment by wesm** [comment]
+1 -- @xhochy if you run into any issues around this pls let me know and I will make it a priority to figure out what's up

**Comment by xhochy** [comment]
@wesm This actually broke my `pip install -e .` Before the change it worked directly, now I need to set `LD_LIBRARY_PATH`  to the location of the arrow-cpp installation

**Comment by wesm** [comment]
This is the correct behavior for relocatability, I think, otherwise you have a hard-coded rpath in the shared libraries.

**Pull request closed without merging**